### DISCUSSION TITLE
Timestamp

### DISCUSSION THREAD

**Discussion by murtaghj** [aminalaee/uuid-utils#20] [Q&A] [answered]
Hey, I’m considering going with uuid7 for a database primary key. I know it won’t be as efficient as as auto number due to the size etc but I think this is I’ll help solve the clustered index issue. 

I tried to extract the timestamp from the generated uuid. Does the time method return the ns since epoch? The values I’m seeing on my machine are kind of strange and don’t seem to represent the current time. Maybe I’m just missing something? Any help much appreciated.

**Comment by aminalaee** [comment, marked as answer]
Hey @murtaghj ,
I think you can use the following snippet to get it:

```py
import uuid_utils
b = uuid_utils.uuid7().bytes

millis = b[0] << 40 | b[1] << 32 | b[2] << 24 | b[3] << 16 | b[4] << 8 | b[5]
print(millis)
```

I think this is a useful method, I will add it to the public methods to make it easier with `uuid_utils.uuid7().timestamp` or something

**Reply by murtaghj** [reply]
Nice thanks! Yeh I think this would be really useful since it would mean not having to store a created time field since I can derive it.

**Comment by aminalaee** [comment]
Added it in https://github.com/aminalaee/uuid-utils/pull/22
